src/inpod/statemanager.rs

```diff
@@ -172,12 +173,12 @@ impl WorkloadProxyManagerState {
                     "pod delete request, shutting down proxy"
                 );
                 if !self.snapshot_received {
-                    // TODO: consider if this is an error. if not, do this instead:
-                    // self.snapshot_names.remove(&workload_uid)
-                    // self.pending_workloads.remove(&workload_uid)
-                    return Err(Error::ProtocolError(
-                        "pod delete received before snapshot".into(),
-                    ));
+                    debug!("got workload delete before snapshot");
+                    // Since we insert here on AddWorkload before we get a snapshot,
+                    // make sure we also opportunistically remove here before we
+                    // get a snapshot
+                    self.snapshot_names.remove(&workload_uid);
+                    return Ok(());
```

Comment:
the todo comment suggested `self.pending_workloads.remove(&workload_uid)`.. and reason we don't need to do that?

Reply:
Ah good catch, we do, I was originally letting this fall through to the `del_workload()`.

Reply:
Oh actually - now I remember. When we get the `SnapshotDone` we would auto-drop anything that had previously been started but wasn't in the snapshot, so it _should_ get handled there.

Reply:
That being said, explicitly removing at-time-of-message is probably _simpler_  and safer so I'll just do that.

Reply:
Settled on a clarifying comment.

Reply:
...but we can't do it in `reconcile()`, since `workload_states` only has successfully started proxies.

Updated and added comments to hopefully make this more clear.